tests/unit/s2n_tls13_pq_handshake_test.c

```diff
@@ -590,6 +585,8 @@ int main()
             } else {
                 curve = expected_curve;
             }
+        } else if (kem_group != NULL && !kem_group->available) {
+            kem_group = prefers_x25519_over_secp256r1(client_policy) ? &s2n_x25519_kyber_512_r3 : &s2n_secp256r1_kyber_512_r3;
```

Comment:
It seems a bit odd to override the expected kem group value from the test case. I wonder if the kem group field in the test case should be updated instead? So maybe change what `expected_kyber_r3_group` is set to? Or maybe the expected kem group should always just be decided here, rather than in the test cases? Or maybe the current way makes sense if it's a lot easier this way.

Reply:
yeah, initially i'd tried to simply set `expected_kem_group` appropriately, but that gets tricky because the preference depends on whether underlying libcrypto supports x25519. So, we'd need a runtime call to `prefers_x25519_over_secp256r1` when declaring the `test_vectors[]` array but you can't inline the runtime call in a stack-allocated array declaration. We could do that with some ifdefs, but one of the goals of this PR is to reduce reliance on conditional compilation.

the current isn't particularly elegant, but I believe it's better than the alternatives.

Reply:
> then the expected kem could be decided with s2n_get_highest_priority_kem_group from the client's kem preferences if the expected_curve field is NULL.

Why would we pull this from the client's KEM preferences instead of the server's? the expected group would be the highest mutually supported (and available!) KEM group between the client and server security policies, right?

Reply:
I suppose I could write a small function to do this. That wouldn't be terrible.

Reply:
Ok, I did this in the latest commit. nice find!

https://github.com/aws/s2n-tls/pull/4101/commits/7507c8b022abbcaae6ed97d72910a48bcf92fb5e